pkg/controller/podgc/gc_controller.go

```diff
@@ -282,7 +287,7 @@ func (gcc *PodGCController) gcUnscheduledTerminating(pods []*v1.Pod) {
 		}
 
 		klog.V(2).InfoS("Found unscheduled terminating Pod not assigned to any Node, deleting.", "pod", klog.KObj(pod))
-		if err := gcc.deletePod(pod.Namespace, pod.Name); err != nil {
+		if err := gcc.deletePod(pod.Namespace, pod.Name, v1.GarbageCollectedWhileTerminating); err != nil {
```

Comment:
I don't think we should add a condition in this case. Something else already removed the Pod (because it has a deletionTimestamp) so it was not a disruption by the system.